mitodl/ocw-studio
# Add name to site creation form, but autopopulate based on title input

Blocked until a final decision is made on whether or not to allow non-ascii urls, not a high priority yet.

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #128 

#### What's this PR do?
Adds a `name` input field to the site creation form, autopopulates it with a slug based on the `title` field input, and prevents non-ascii characters from being used.

#### How should this be manually tested?
Go to the form for creating a new site.  As you enter a title, the name field below should start to populate.  It should contain only lowercase a-z, 0-9, and - characters.  If you try to add other characters you should see a validation error.  Submit the form, the site should be created with that name/slug.

#### Where should the reviewer start?
Initially thought about allowing non-ascii characters in the name by passing `allow_unicode=True` to `django.utils.text.slugify`, but after talking with Ferdi & Nathan decided to disallow it and use this approach instead.

#### Screenshots
<img width="742" alt="Screen Shot 2021-06-14 at 3 22 36 PM" src="https://user-images.githubusercontent.com/187676/121947773-7166a080-cd24-11eb-8ec1-d0661110ef7e.png">
